tests/zip_function.test.ts

```diff
@@ -138,4 +138,34 @@ describe('zipFunction', () => {
       expect(mock2).toBe(true)
     },
   )
+
+  testMany(
+    'Can populate the isInternalFunction property for functions',
+    ['bundler_default', 'bundler_esbuild', 'bundler_nft'],
+    async (options) => {
+      const { path: tmpDir } = await getTmpDir({ prefix: 'zip-it-test' })
+      const basePath = join(FIXTURES_DIR, 'node-internal', '.netlify/internal-functions')
+      const opts = merge(options, {
+        internalFunctionsFolder: basePath,
+      })
+      const result = (await zipFunction(`${basePath}/function-1.js`, tmpDir, opts))!
+
+      expect(result.isInternalFunction).toBeTruthy()
```

Comment:
`toBeTruthy` will accept any value that isn't falsy (🥁). It's not guaranteeing that `result.isInternalFunction` is `true`, which I think we should be guaranteeing here.

Reply:
Makes sense! I'll change it :)